### ISSUE TITLE
ERROR: type NamedTuple has no field ...

### ISSUE THREAD

**Issue by yakir12** [JuliaComputing/JuliaDB.jl#291]
After many attempts to debug this on slack#juliadb (thanks Julian Samaroo and Pietro Vertechi) I really don't know what is going on. Following is a MWE of joining two tables. 

```julia
using DataDeps, JuliaDB
register(DataDep("coffeebeetle2", "the coffee beetle database", "https://s3.eu-central-1.amazonaws.com/vision-group-file-sharing/Data%20backup%20and%20storage/Yakir/coffee%20beetles/database.zip", "557ec5027d34d2641db4f2cc61c993edd2ce2bee1530254d45d26cc41c5adf7a", post_fetch_method = unpack))
board = loadtable(datadep"coffeebeetle2/board.csv", indexcols = :designation)
calibration = loadtable(datadep"coffeebeetle2/calibration.csv", indexcols = :calibration)
x = join(calibration, board, lkey = :board, rkey = :designation)
```
and the resulting error:
```julia
ERROR: type NamedTuple has no field board
Stacktrace:
 [1] #134 at ./none:0 [inlined]
 [2] iterate(::Base.Generator{Tuple{Symbol},getfield(Base, Symbol("##134#137")){NamedTuple{(:designation,),Tuple{PooledArrays.PooledArray{String,UInt8,1,Array{UInt8,1}}}}}}) at ./generator.jl:47
 [3] append_any(::Any, ::Vararg{Any,N} where N) at ./essentials.jl:674
 [4] (::Type{NamedTuple{names,T} where T<:Tuple})(::NamedTuple) at ./namedtuple.jl:87
 [5] StructArrays.StructArray{NamedTuple{(:board,),Tuple{String}},N,C,I} where I where C<:Union{Tuple, NamedTuple} where N(::NamedTuple{(:designation,),Tuple{PooledArrays.PooledArray{String,UInt8,1,Array{UInt8,1}}}}) at /home/yakir/.julia/packages/StructArrays/RkrVr/src/structarray.jl:31
 [6] #call#7 at /home/yakir/.julia/packages/StructArrays/RkrVr/src/structarray.jl:48 [inlined]
 [7] StructArrays.StructArray{NamedTuple{(:board,),Tuple{String}},1,C,I} where I where C<:Union{Tuple, NamedTuple}(::NamedTuple{(:designation,),Tuple{PooledArrays.PooledArray{String,UInt8,1,Array{UInt8,1}}}}) at /home/yakir/.julia/packages/StructArrays/RkrVr/src/structarray.jl:48
 [8] #join#276(::Symbol, ::Bool, ::Symbol, ::Symbol, ::NTuple{4,Int64}, ::NTuple{4,Int64}, ::Nothing, ::Bool, ::Type, ::Nothing, ::Nothing, ::typeof(join), ::typeof(IndexedTables.concat_tup), ::IndexedTable{StructArrays.StructArray{NamedTuple{(:calibration, :intrinsic, :extrinsic, :board, :comment),NTuple{5,String}},1,NamedTuple{(:calibration, :intrinsic, :extrinsic, :board, :comment),NTuple{5,WeakRefStrings.StringArray{String,1}}},Int64}}, ::IndexedTable{StructArrays.StructArray{NamedTuple{(:designation, :checker_width_cm, :checker_per_width, :checker_per_height, :board_description),Tuple{String,Float64,Int64,Int64,String}},1,NamedTuple{(:designation, :checker_width_cm, :checker_per_width, :checker_per_height, :board_description),Tuple{WeakRefStrings.StringArray{String,1},Array{Float64,1},Array{Int64,1},Array{Int64,1},WeakRefStrings.StringArray{String,1}}},Int64}}) at /home/yakir/.julia/packages/IndexedTables/DgX0A/src/join.jl:182
 [9] (::getfield(Base, Symbol("#kw##join")))(::NamedTuple{(:how, :lkey, :rkey),Tuple{Symbol,Symbol,Symbol}}, ::typeof(join), ::Function, ::IndexedTable{StructArrays.StructArray{NamedTuple{(:calibration, :intrinsic, :extrinsic, :board, :comment),NTuple{5,String}},1,NamedTuple{(:calibration, :intrinsic, :extrinsic, :board, :comment),NTuple{5,WeakRefStrings.StringArray{String,1}}},Int64}}, ::IndexedTable{StructArrays.StructArray{NamedTuple{(:designation, :checker_width_cm, :checker_per_width, :checker_per_height, :board_description),Tuple{String,Float64,Int64,Int64,String}},1,NamedTuple{(:designation, :checker_width_cm, :checker_per_width, :checker_per_height, :board_description),Tuple{WeakRefStrings.StringArray{String,1},Array{Float64,1},Array{Int64,1},Array{Int64,1},WeakRefStrings.StringArray{String,1}}},Int64}}) at ./none:0
 [10] #join#277 at /home/yakir/.julia/packages/IndexedTables/DgX0A/src/join.jl:197 [inlined]
 [11] (::getfield(Base, Symbol("#kw##join")))(::NamedTuple{(:lkey, :rkey),Tuple{Symbol,Symbol}}, ::typeof(join), ::IndexedTable{StructArrays.StructArray{NamedTuple{(:calibration, :intrinsic, :extrinsic, :board, :comment),NTuple{5,String}},1,NamedTuple{(:calibration, :intrinsic, :extrinsic, :board, :comment),NTuple{5,WeakRefStrings.StringArray{String,1}}},Int64}}, ::IndexedTable{StructArrays.StructArray{NamedTuple{(:designation, :checker_width_cm, :checker_per_width, :checker_per_height, :board_description),Tuple{String,Float64,Int64,Int64,String}},1,NamedTuple{(:designation, :checker_width_cm, :checker_per_width, :checker_per_height, :board_description),Tuple{WeakRefStrings.StringArray{String,1},Array{Float64,1},Array{Int64,1},Array{Int64,1},WeakRefStrings.StringArray{String,1}}},Int64}}) at ./none:0
 [12] top-level scope at none:0
```
I'm on Julia 1.1.1 and the status --manifest is:
```
(v1.1) pkg> st --manifest JuliaDB IndexedTables
    Status `~/.julia/environments/v1.1/Manifest.toml`
  [d58978e5] Dagger v0.8.0
  [e7dc6d0d] DataValues v0.4.11
  [c27321d9] Glob v1.2.0
  [6deec6e2] IndexedTables v0.12.0
  [82899510] IteratorInterfaceExtensions v1.0.0
  [a93385a2] JuliaDB v0.12.0
  [f9f48841] MemPool v0.2.0
  [4d1e1d77] Nullables v0.0.8
  [a15396b6] OnlineStats v0.23.0
  [925886fa] OnlineStatsBase v0.10.2
  [2dfb63ee] PooledArrays v0.5.2
  [3cdcf5f2] RecipesBase v0.6.0
  [2913bbd2] StatsBase v0.30.0
  [09ab397b] StructArrays v0.4.0
  [3783bdb8] TableTraits v1.0.0
  [382cd787] TableTraitsUtils v1.0.0
  [bd369af6] Tables v0.2.8
  [e0df1984] TextParse v0.9.1
  [ea10d353] WeakRefStrings v0.5.8
  [ade2ca70] Dates 
  [8ba89e20] Distributed 
  [de0858da] Printf 
  [9a3f8284] Random 
  [9e88b42a] Serialization 
  [2f01184e] SparseArrays 
  [10745b16] Statistics 
  [8dfed614] Test 
```

**Comment by jpsamaroo** [comment]
It appears that IndexedTables `v0.12.1` has an upper bound in the registry on StructArrays `v0.3.4`, but IndexedTables `v0.12.0` has no such bounds. I noticed this because Pkg upgraded only StructArrays to `v0.4.0` when I pinned IndexedTables to `v0.12.0`, and it's these set of versions that allow me to reproduce this issue. This is not, however, reproducible for me on IndexedTables `v0.12.1` and StructArrays `v0.3.4`.
